static/app/views/performance/newTraceDetails/index.tsx

```diff
@@ -331,12 +334,20 @@ export function TraceViewWaterfall(props: TraceViewWaterfallProps) {
       return loadingTrace;
     }
 
-    if (props.trace) {
-      return TraceTree.FromTrace(props.trace, props.replayRecord);
+    if (traceData) {
+      return TraceTree.FromTrace(traceData, props.replayRecord, {
+        api,
+        filters,
+        additionalTraceDataRows: props.traceDataRows?.slice(1),
+        organization: props.organization,
+        traceLimit: undefined,
+        urlParams,
+        rerender: forceRerender,
+      });
```

Comment:
There is no need to put this in a constructor. We should construct the tree in the same way as we did before with a minimal set of arguments needed to create a valid trace tree and call a separate method later on to fetch this.

Reply:
Hey, @JonasBa I pulled out the incremental logic from the constructor and added a function called `incrementallyFetchTraces` that does the job.